web/src/app/services/data-store/data-store.service.ts

```diff
@@ -312,34 +313,68 @@ export class DataStoreService {
   }
 
   lois$({id}: Survey, {email}: User): Observable<List<LocationOfInterest>> {
-    const predefinedLois = this.db.collection(
-      `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
-      ref => ref.where('predefined', '==', true)
-    );
-
-    const userLois = this.db.collection(
-      `${SURVEYS_COLLECTION_NAME}/${id}/lois`,
-      ref =>
+    return this.db
```

Comment:
This method has become unwieldy. Please consider refactoring the creation of the different streams into their own methods.